app/forms/validators.py

```diff
@@ -417,8 +417,8 @@ def __call__(
             condition = f"{conditions[0]} or equals"
         else:
             condition = conditions[0]
-
-        is_valid, message = self._is_valid(condition, total, target_total)
+        # is_valid never returns a "None" object. Therefore a none object being iterated will never occur
+        is_valid, message = self._is_valid(condition, total, target_total)  # type: ignore[misc]
```

Comment:
What is the use of `[misc]`

Reply:
Mypy performs numerous other, less commonly failing checks that don’t have specific error codes. They are labeled as miscs. This one suggested` is_valid` fails to iterate through a `None object`, but I believe `is_valid `should never be None.

Reply:
It only occurred when I was adding Optional type hinting due to it being also "potentially" a None I can have a look through to see if it can be fixed.

Reply:
I cannot seem to find a solution so I believe in keeping `type ignore[misc]`. We should leave`[misc]` specifically because mypy can still check for other typing errors on that line, and with new updates to mypy certain `misc` errors will get their own specific error message.

Reply:
This is a genuine warning. `_is_valid` could return None and this would result in an error. It never does at the moment because we don't have another other condition. 

This can be fixed by updating that method to raise a `NotImplementedError` and should always return the tuple.

Reply:
Changes added, thanks for the help on this one. It makes a lot more sense now